test/lib/ufe/testAttributeBlock.py

```diff
@@ -0,0 +1,178 @@
+        rootLayer = stage.GetRootLayer()
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a transform3d
+        cylinderPath = ufe.Path([
+                     mayaUtils.createUfePathSegment("|mayaUsdTransform|shape"), 
+                     usdUtils.createUfePathSegment("/pCylinder1")])
+        cylinderItem = ufe.Hierarchy.createItem(cylinderPath)
+
+        cylinderT3d = ufe.Transform3d.transform3d(cylinderItem)
+
+        # create a xformable
+        cylinderXformable = UsdGeom.Xformable(cylinderPrim)
+
+        # writing to "transform op order" is expected to be allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(cylinderXformable.GetXformOpOrderAttr()))
+
+        # do any transform editing.
+        cylinderT3d = ufe.Transform3d.transform3d(cylinderItem)
+        cylinderT3d.scale(2.5, 2.5, 2.5)
+        cylinderT3d.translate(4.0, 2.0, 3.0)
+
+        # check the "transform op order" stack.
+        self.assertEqual(cylinderXformable.GetXformOpOrderAttr().Get(), Vt.TokenArray(('xformOp:translate', 'xformOp:scale')))
+
+        # check if translate attribute is editable
+        translateAttr = cylinderPrim.GetAttribute('xformOp:translate')
+        self.assertIsNotNone(translateAttr)
+
+        # authoring new transformation edit is expected to be allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(translateAttr))
+        cylinderT3d.translate(5.0, 6.0, 7.0)
+        self.assertEqual(translateAttr.Get(), Gf.Vec3d(5.0, 6.0, 7.0))
+
+        # set the edit target to a weaker layer (LayerC)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerC)
+
+        # authoring new transformation edit is not allowed.
+        self.assertFalse(mayaUsdUfe.isAttributeEditAllowed(translateAttr))
+
+        # set the edit target to a stronger layer (LayerA)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerA)
+
+        # authoring new transformation edit is allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(translateAttr))
+        cylinderT3d.rotate(0.0, 90.0, 0.0)
+
+        # check the "transform op order" stack.
+        self.assertEqual(cylinderXformable.GetXformOpOrderAttr().Get(), Vt.TokenArray(('xformOp:translate','xformOp:rotateXYZ', 'xformOp:scale')))
+
+if __name__ == '__main__':
+    unittest.main(verbosity=2)
```

Comment:
@ppt-adsk 

My original plan was to have this test as part of existing testAttribute.py but then realized that it only runs with UFE2.0

https://github.com/Autodesk/maya-usd/blob/575bcc6528d3cb5da1e1d5edc5291e34fcb663a1/test/lib/ufe/CMakeLists.txt#L20

I could expose testAttribute to run by both UFE1.0 and UFE2.0 but that means I have to isolate various parts of the codes which could be very messy considering the UFE1.0 doesn't support a lot of great features that UFE2.0 has, etc....

I was thinking that perhaps I could move "testSingleAttributeBlocking" and "testTransformationAttributeBlocking" from 
testAttribute.py to testAttributeBlock.py and wrap certain parts of the logics around different UFE versions ( e.g ufe.PathString , contexOpsMenu, etc... )

What do you think?